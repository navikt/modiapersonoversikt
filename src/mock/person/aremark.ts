import { Diskresjonskoder, TilrettelagtKommunikasjonsTyper } from '../../konstanter';
import { Kjønn, Person, Relasjonstype, SivilstandTyper } from '../../models/person/person';

export const aremark: Person = {
    fødselsnummer: '10108000398',
    kjønn: Kjønn.Mann,
    geografiskTilknytning: '0118',
    alder: 42,
    navn: {
        sammensatt: 'AREMARK TESTFAMILIEN',
        fornavn: 'AREMARK',
        mellomnavn: '',
        etternavn: 'TESTFAMILIEN',
    },
    tilrettelagtKomunikasjonsListe: [
        {
            behovKode: 'TOHJ',
            beskrivelse: TilrettelagtKommunikasjonsTyper.TOHJ
        }
    ],
    diskresjonskode: Diskresjonskoder.FORTROLIG_ADRESSE,
    statsborgerskap: 'NORSK',
    personstatus: {
        dødsdato: undefined,
        bostatus: undefined
    },
    sivilstand: {
        value: SivilstandTyper.Gift,
        beskrivelse: 'Gift',
        fraOgMed: '2005-12-12'
    },
    familierelasjoner: [{
        harSammeBosted: false,
        rolle: Relasjonstype.Barn,
        tilPerson: {
            navn: {
                sammensatt: '',
                fornavn: 'Siri',
                mellomnavn: null,
                etternavn: 'Aremark'
            },
            alder: 17,
            fødselsnummer: '10117100000',
            personstatus: {}
        }

    }],
    kontaktinformasjon: {
        mobil: {
            sistEndret: '2014-06-21T18:44:39+02:00',
            telefonnummer: '99887766',
            retningsnummer: '47',
            sistEndretAv: 'BRUKER'
        }
    }
};