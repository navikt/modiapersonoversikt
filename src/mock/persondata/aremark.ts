import {
    AdresseBeskyttelse,
    EgenAnsatt,
    ForelderBarnRelasjon,
    ForelderBarnRelasjonRolle,
    FullmaktsRolle,
    Kjonn,
    LocalDate,
    LocalDateTime,
    Person,
    PersonStatus,
    SivilstandType,
    Skifteform
} from '../../app/personside/visittkort-v2/PersondataDomain';
import { harDiskresjonskode } from '../../app/personside/visittkort-v2/visittkort-utils';

export function hentBarnAremark(): ForelderBarnRelasjon[] {
    return [
        {
            ident: '12345678910',
            rolle: ForelderBarnRelasjonRolle.BARN,
            navn: [
                {
                    fornavn: 'BARN1',
                    mellomnavn: null,
                    etternavn: 'BARNESEN'
                }
            ],
            fodselsdato: ['2008-03-15' as LocalDate],
            alder: 13,
            kjonn: [
                {
                    kode: Kjonn.K,
                    beskrivelse: 'Kvinne'
                }
            ],
            adressebeskyttelse: [
                {
                    kode: AdresseBeskyttelse.UGRADERT,
                    beskrivelse: 'UGRADERT'
                }
            ],
            harSammeAdresse: false,
            dodsdato: []
        },
        {
            ident: '12345678911',
            rolle: ForelderBarnRelasjonRolle.BARN,
            navn: [
                {
                    fornavn: 'BARN1',
                    mellomnavn: null,
                    etternavn: 'BARNESEN'
                }
            ],
            fodselsdato: ['2008-03-15' as LocalDate],
            alder: null,
            kjonn: [
                {
                    kode: Kjonn.K,
                    beskrivelse: 'Kvinne'
                }
            ],
            adressebeskyttelse: [],
            harSammeAdresse: false,
            dodsdato: []
        }
    ];
}

export const aremark: Person = {
    fnr: '10108000398',
    navn: [
        {
            fornavn: 'AREMARK',
            mellomnavn: null,
            etternavn: 'TESTFAMILIEN'
        }
    ],
    kjonn: [
        {
            kode: Kjonn.M,
            beskrivelse: 'Mann'
        }
    ],
    fodselsdato: ['1980-10-10' as LocalDate],
    geografiskTilknytning: '0118',
    alder: 41,
    dodsdato: [],
    bostedAdresse: [
        {
            linje1: 'Islandsgate 49',
            linje2: '7693 Hovedøya',
            linje3: 'Norge',
            sistEndret: {
                ident: 'Folkeregisteret',
                tidspunkt: '2010-02-20T10:15:30' as LocalDateTime,
                system: 'FREG'
            },
            gyldighetsPeriode: null
        }
    ],
    kontaktAdresse: [
        {
            linje1: 'Aremarkveien 1',
            linje2: '0320 Aremark',
            linje3: null,
            sistEndret: {
                ident: 'Folkeregisteret',
                tidspunkt: '2020-01-01T10:15:30' as LocalDateTime,
                system: 'FREG'
            },
            gyldighetsPeriode: null
        }
    ],
    oppholdsAdresse: [
        {
            linje1: 'Oppholdsadresse 1',
            linje2: '0000 AREMARK',
            linje3: null,
            sistEndret: null,
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-02-01' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            }
        }
    ],
    navEnhet: {
        id: '0219',
        navn: 'NAV Bærum',
        publikumsmottak: [
            {
                besoksadresse: {
                    linje1: 'Adressevei 1',
                    linje2: '0000 AREMARK',
                    linje3: null,
                    sistEndret: null,
                    gyldighetsPeriode: null
                },
                apningstider: [
                    {
                        ukedag: 'Mandag',
                        apningstid: '09.00 - 15.00'
                    },
                    {
                        ukedag: 'Tirsdag',
                        apningstid: '09.00 - 15.00'
                    },
                    {
                        ukedag: 'Onsdag',
                        apningstid: '10.00 - 16.00'
                    }
                ]
            }
        ]
    },
    statsborgerskap: [
        {
            land: {
                kode: 'NOR',
                beskrivelse: 'NORGE'
            },
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-02-01' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            }
        }
    ],
    adressebeskyttelse: [
        {
            kode: AdresseBeskyttelse.KODE6,
            beskrivelse: 'Kode 6'
        }
    ],
    sikkerhetstiltak: [
        {
            type: 'TFUS',
            beskrivelse: 'Telefonisk utestengelse',
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-02-01' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            }
        }
    ],
    erEgenAnsatt: EgenAnsatt.JA,
    personstatus: [
        {
            kode: PersonStatus.BOSATT,
            beskrivelse: 'Bosatt i Norge'
        }
    ],
    sivilstand: [
        {
            type: {
                kode: SivilstandType.GIFT,
                beskrivelse: 'Gift'
            },
            gyldigFraOgMed: '2005-12-12' as LocalDate,
            sivilstandRelasjon: {
                fnr: '12345555555',
                navn: [
                    {
                        fornavn: 'BAREMARK',
                        mellomnavn: null,
                        etternavn: 'TESTFAMILIEN'
                    }
                ],
                alder: 40,
                adressebeskyttelse: [
                    {
                        kode: AdresseBeskyttelse.UGRADERT,
                        beskrivelse: 'UGRADERT'
                    }
                ],
                harSammeAdresse: true
            }
        }
    ],
    foreldreansvar: hentBarnAremark().map((barn) => ({
        ansvar: 'felles',
        ansvarlig: null,
        ansvarsubject: {
            navn: harDiskresjonskode(barn.adressebeskyttelse) ? null : barn.navn.firstOrNull(),
            ident: barn.ident
        }
    })),
    deltBosted: [
        {
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-02-01' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            },
            adresse: {
                linje1: `Adresseveien 1`,
                linje2: '0000 Aremark',
                linje3: null,
                sistEndret: null,
                gyldighetsPeriode: null
            }
        }
    ],
    dodsbo: [
        {
            adressat: {
                advokatSomAdressat: {
                    kontaktperson: {
                        fornavn: 'Advokat',
                        mellomnavn: null,
                        etternavn: 'Navnesen'
                    },
                    organisasjonsnavn: 'Advokatkontoret Aremark',
                    organisasjonsnummer: '01234567891011'
                },
                organisasjonSomAdressat: null,
                personSomAdressat: null
            },
            adresse: {
                linje1: 'Elgelia 20, 0000 Aremark',
                linje2: null,
                linje3: null,
                sistEndret: null,
                gyldighetsPeriode: null
            },
            registrert: '2010-02-02' as LocalDate,
            skifteform: Skifteform.OFFENTLIG,
            sistEndret: {
                ident: 'Folkeregisteret',
                tidspunkt: '2020-01-01T10:15:30' as LocalDateTime,
                system: 'Folkeregisteret'
            }
        }
    ],
    fullmakt: [
        {
            motpartsPersonident: '123456789',
            motpartsPersonNavn: {
                fornavn: 'Navn',
                mellomnavn: null,
                etternavn: 'Navnesen'
            },
            motpartsRolle: FullmaktsRolle.FULLMEKTIG,
            omrade: [
                {
                    kode: '*',
                    beskrivelse: ''
                }
            ],
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-06-01' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            }
        }
    ],
    vergemal: [
        {
            ident: '21042900076',
            navn: {
                fornavn: 'Simen',
                mellomnavn: null,
                etternavn: 'Solli'
            },
            vergesakstype: 'Voksen',
            omfang: 'Ivareta personens interesser innenfor det personlige og økonomiske området',
            embete: 'Fylkesmannen i Troms og Finnmark',
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-06-01' as LocalDate,
                gyldigTilOgMed: '2022-05-10' as LocalDate
            }
        },
        {
            ident: '123456799',
            navn: {
                fornavn: 'Truls',
                mellomnavn: null,
                etternavn: 'Tøffel'
            },
            vergesakstype: 'Fremtidsfullmakt',
            omfang: 'Ivareta personens interesser innenfor det økonomiske området',
            embete: 'Fylkesmannen i Troms og Finnmark',
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-02-10' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            }
        }
    ],
    tilrettelagtKommunikasjon: {
        talesprak: [
            {
                kode: 'NO',
                beskrivelse: 'Norsk'
            }
        ],
        tegnsprak: [
            {
                kode: 'NO',
                beskrivelse: 'Norsk'
            }
        ]
    },
    telefonnummer: [
        {
            retningsnummer: {
                kode: '+47',
                beskrivelse: 'Norge'
            },
            identifikator: '99009900',
            sistEndret: {
                ident: '11223344',
                tidspunkt: '2018-06-14T00:00:00' as LocalDateTime,
                system: 'BD06'
            },
            prioritet: -1
        }
    ],
    kontaktOgReservasjon: {
        personident: '10108000398',
        reservasjon: 'true',
        epostadresse: {
            value: 'epost@nav.no',
            sistOppdatert: null,
            sistVerifisert: '2013-01-01' as LocalDate
        },
        mobiltelefonnummer: {
            value: '12345678',
            sistOppdatert: '2021-11-02' as LocalDate,
            sistVerifisert: '2021-11-02' as LocalDate
        }
    },
    bankkonto: {
        kontonummer: '12345678900',
        banknavn: 'DNB ASA',
        sistEndret: {
            ident: '1010800 BD03',
            tidspunkt: '2006-03-15T00:00:00' as LocalDateTime,
            system: ''
        },
        bankkode: null,
        swift: 'DNBANOKKXXX',
        landkode: null,
        adresse: {
            linje1: 'Bankveien 1,',
            linje2: '0357 Bankestad',
            linje3: null,
            sistEndret: null,
            gyldighetsPeriode: null
        },
        valuta: {
            kode: 'NOK',
            beskrivelse: 'Norske kroner'
        }
    },
    forelderBarnRelasjon: hentBarnAremark()
};
