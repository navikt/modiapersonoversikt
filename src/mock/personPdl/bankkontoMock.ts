import faker from 'faker/locale/nb_NO';

import { Bankkonto } from '../../models/personPdl/bankkonto';
import { getSistOppdatert, vektetSjanse } from '../utils/mock-utils';

export const bankkontoNorsk: Bankkonto = {
    kontonummer: Number(faker.finance.account(11)).toString(),
    banknavn: 'Nordea ASA',
    sistEndret: new Date(getSistOppdatert()),
    sistEndretAv: '1010800 BD03',

    bankkode: null,
    swift: faker.finance.bic(),
    landkode: null,
    adresse: {
        linje1: faker.address.streetAddress(),
        linje2: faker.address.zipCode('####') + faker.address.county(),
        linje3: null
    },
    valuta: {
        kode: 'NOK',
        beskrivelse: 'Norske kroner'
    }
};

export const bankkontoUtland: Bankkonto = {
    kontonummer: Number(faker.finance.account(9)).toString(),
    banknavn: 'BBVA',
    sistEndret: new Date(getSistOppdatert()),
    sistEndretAv: '1010800 BD03',
    swift: faker.finance.bic(),
    bankkode: null,
    landkode: {
        kode: faker.address.countryCode(),
        beskrivelse: faker.address.county()
    },
    adresse: {
        linje1: faker.address.streetAddress(),
        linje2: faker.address.zipCode('##### ') + faker.address.county(),
        linje3: null
    },
    valuta: {
        kode: faker.finance.currencyCode(),
        beskrivelse: faker.finance.currencyName()
    }
};

export function getBankKonto(): Bankkonto | null {
    if (vektetSjanse(faker, 0.7)) {
        return bankkontoNorsk;
    } else if (vektetSjanse(faker, 0.2)) {
        return bankkontoUtland;
    } else {
        return null;
    }
}
