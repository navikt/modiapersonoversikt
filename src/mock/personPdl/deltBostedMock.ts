import { DeltBosted } from '../../models/personPdl/deltBosted';

export function getDeltBostedMock(faker: Faker.FakerStatic): DeltBosted[] {
    return [
        {
            startdatoForKontrakt: new Date(faker.date.past(1)),
            sluttdatoForKontrakt: new Date(faker.date.future(2)),
            adresse: {
                linje1: `${faker.address.streetAddress()} ${faker.address.city()} ${faker.address.zipCode('####')}`,
                linje2: null,
                linje3: null
            }
        }
    ];
}
