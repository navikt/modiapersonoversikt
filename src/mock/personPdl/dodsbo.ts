import { fyllRandomListe, vektetSjanse } from '../utils/mock-utils';
import {
    Dodsbo,
    Adressat,
    AdvokatSomAdressat,
    PersonSomAdressat,
    OrganisasjonSomAdressat,
    Skifteform
} from '../../models/personPdl/dodsbo';
import { Navn } from '../../models/personPdl/person';

export function getMockDodsbo(faker: Faker.FakerStatic): Dodsbo[] {
    return fyllRandomListe<Dodsbo>(() => mockDodsbo(faker), 2, false);
}

function mockDodsbo(faker: Faker.FakerStatic): Dodsbo {
    return {
        adressat: mockAdressat(faker),
        adresse: {
            linje1: `${faker.address.streetAddress()} ${faker.address.city()} ${faker.address.zipCode('####')}`,
            linje2: null,
            linje3: null
        },
        registrert: new Date('2020-02-02'),
        skifteform: Skifteform.Offentlig
    };
}

function mockAdressat(faker: Faker.FakerStatic): Adressat {
    if (vektetSjanse(faker, 0.2)) {
        return {
            advokatSomAdressat: mockAdvokatSomAdressat(faker),
            organisasjonSomAdressat: null,
            personSomAdressat: null
        };
    } else if (vektetSjanse(faker, 0.2)) {
        return {
            organisasjonSomAdressat: mockOrganisasjonSomAdressat(faker),
            advokatSomAdressat: null,
            personSomAdressat: null
        };
    } else {
        return {
            advokatSomAdressat: null,
            organisasjonSomAdressat: null,
            personSomAdressat: mockPersonSomAdressat(faker)
        };
    }
}

function mockAdvokatSomAdressat(faker: Faker.FakerStatic): AdvokatSomAdressat {
    return {
        kontaktperson: mockPersonnavn(faker),
        organisasjonsnavn: faker.company.companyName(),
        organisasjonsnummer: `${faker.random.number()}`
    };
}

function mockOrganisasjonSomAdressat(faker: Faker.FakerStatic): OrganisasjonSomAdressat {
    return {
        kontaktperson: mockPersonnavn(faker),
        organisasjonsnavn: faker.company.companyName(),
        organisasjonsnummer: `${faker.random.number()}`
    };
}

function mockPersonSomAdressat(faker: Faker.FakerStatic): PersonSomAdressat {
    return {
        fnr: '10.10.1980',
        navn: mockPersonnavn(faker),
        fodselsdato: new Date('1980-10-10')
    };
}

export function mockPersonnavn(faker: Faker.FakerStatic): Navn {
    return {
        fornavn: faker.name.firstName(),
        etternavn: faker.name.lastName(),
        mellomnavn: null
    };
}
