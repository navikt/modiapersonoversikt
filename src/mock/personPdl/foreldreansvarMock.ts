import { aremark } from './aremark';
import faker from 'faker/locale/nb_NO';
import navfaker from 'nav-faker';
import { vektetSjanse } from '../utils/mock-utils';
import { Foreldreansvar } from '../../models/personPdl/foreldreansvar';
import { mockPersonnavn } from './dodsbo';

export function mockVergemal(fodselsnummer: string) {
    if (fodselsnummer === aremark.fnr) {
        return getAremarkForeldreansvar();
    }
    faker.seed(Number(fodselsnummer));
    navfaker.seed(fodselsnummer);
    if (vektetSjanse(faker, 0.7)) {
        return {};
    }
    return getForeldreansvar();
}

function getForeldreansvar(): Foreldreansvar {
    return {
        ansvar: getTilfeldigAnsvar(),
        ansvarlig: mockPersonnavn(faker),
        ansvarsubject: mockPersonnavn(faker)
    };
}

export function getAremarkForeldreansvar(): Foreldreansvar {
    return {
        ansvar: 'felles',
        ansvarlig: {
            fornavn: 'Test',
            etternavn: 'Testesen',
            mellomnavn: null
        },
        ansvarsubject: {
            fornavn: 'Barn',
            etternavn: 'Barnesen',
            mellomnavn: null
        }
    };
}

function getTilfeldigAnsvar(): string {
    return ANSVARSTYPER[faker.random.number(ANSVARSTYPER.length - 1)];
}

const ANSVARSTYPER = ['felles', 'mor', 'far', 'medmor', 'medfar', 'andre', 'ukjent'];
