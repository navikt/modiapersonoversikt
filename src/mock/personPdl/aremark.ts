import {
    Adresse,
    AdresseBeskyttelse,
    EgenAnsatt,
    Enhet,
    Kjonn,
    KodeBeskrivelse,
    Navn,
    Person,
    PersonStatus,
    Sikkerhetstiltak,
    SikkerhetstiltakType,
    Sivilstand,
    SivilstandType,
    Statsborgerskap,
    Telefon,
    TilrettelagtKommunikasjon
} from '../../models/personPdl/person';
import { getAremarkForeldreansvar } from './foreldreansvarMock';
import { getDeltBostedMock } from './deltBostedMock';
import * as faker from 'faker';
import { getMockDodsbo } from './dodsbo';
import navfaker from 'nav-faker/dist/index';
import { mockVergemal } from './vergemalMock';
import { DigitalKontaktinformasjon } from '../../models/personPdl/digitalKontaktinformasjon';
import { getBankKonto } from './bankkontoMock';
import { Fullmakt, FullmaktsRolle } from '../../models/personPdl/fullmakt';

export const navnAremark: Navn = {
    fornavn: 'TESTFAMILIEN',
    mellomnavn: null,
    etternavn: 'AREMARK'
};

export const kodeBeskrivelseKjonn: KodeBeskrivelse<Kjonn> = {
    kode: Kjonn.Mann,
    beskrivelse: 'Mann'
};

export const adresseAremark: Adresse = {
    linje1: 'Adresseveien 1, 0661 AREMARKSTAD',
    linje2: null,
    linje3: null
};

export const navEnhetAremark: Enhet = {
    id: '0000',
    navn: 'Nav-kontor'
};

export const statsborgerskapAremark: Statsborgerskap = {
    land: {
        kode: 'NOR',
        beskrivelse: 'Norsk statsborger'
    },
    gyldigFraOgMed: new Date('1980-10-10'),
    gyldigTilOgMed: null
};

export const adressebeskyttelseAremark: KodeBeskrivelse<AdresseBeskyttelse> = {
    kode: AdresseBeskyttelse.Ugradert,
    beskrivelse: 'Ingen'
};

export const sikkerhetstiltakAremark: Sikkerhetstiltak = {
    type: SikkerhetstiltakType.TelefoniskUtestengelse,
    gyldigFraOgMed: new Date('2005-02-13'),
    gyldigTilOgMed: new Date('2020-02-02')
};

export const personstatusAremark: KodeBeskrivelse<PersonStatus> = {
    kode: PersonStatus.Bosatt,
    beskrivelse: 'Bosatt i Norge'
};

export const sivilstandAremark: Sivilstand = {
    type: {
        kode: SivilstandType.Skilt,
        beskrivelse: 'Skilt'
    },
    gyldigFraOgMed: new Date('2010-03-06')
};

export const tilrettelagtKommunikasjonAremark: TilrettelagtKommunikasjon = {
    talesprak: [
        {
            kode: 'NO',
            beskrivelse: 'Norsk'
        }
    ],
    tegnsprak: [
        {
            kode: 'NO',
            beskrivelse: 'Norsk'
        }
    ]
};

export const telefonAremark: Telefon = {
    retningsnummer: {
        kode: '+47',
        beskrivelse: 'retningsnummer er norsk'
    },
    identifikator: '99009900',
    sistEndret: new Date('2017-10-10'),
    sistEndretAv: 'TESTFAMILIEN AREMARK',
    prioritet: 1
};

export const digitalKontaktInformasjonAremark: DigitalKontaktinformasjon = {
    personident: '10108000398',
    reservasjon: 'reservert',
    epostadresse: {
        value: null,
        sistOppdatert: null,
        sisVerifisert: new Date('2013-01-01')
    },
    mobiltelefonnummer: {
        value: null,
        sistOppdatert: null,
        sisVerifisert: null
    }
};

export const fullmaktAremark: Fullmakt = {
    motpartsPersonident: '123456789',
    motpartsPersonNavn: {
        fornavn: 'Navn',
        mellomnavn: null,
        etternavn: 'Navnesen'
    },
    motpartsRolle: FullmaktsRolle.Fullmektig,
    omraade: ['*'],
    gyldigFraOgMed: new Date('2019-01-01'),
    gyldigTilOgMed: new Date('2022-12-12')
};

export const aremark: Person = {
    fnr: '10108000398',
    navn: [navnAremark],
    kjonn: [kodeBeskrivelseKjonn],
    fodselsdato: [new Date('1980-10-10')],
    dodsdato: [],
    bostedAdresse: [adresseAremark],
    kontaktAdresse: [adresseAremark],
    navEnhet: navEnhetAremark,
    statsborgerskap: [statsborgerskapAremark],
    adressebeskyttelse: [adressebeskyttelseAremark],
    sikkerhetstiltak: [sikkerhetstiltakAremark],
    erEgenAnsatt: EgenAnsatt.Nei,
    personstatus: [personstatusAremark],
    sivilstand: [sivilstandAremark],
    foreldreansvar: [getAremarkForeldreansvar()],
    deltBosted: getDeltBostedMock(faker),
    dodsbo: getMockDodsbo(faker),
    fullmakt: [fullmaktAremark],
    vergemal: mockVergemal(navfaker.personIdentifikator.f√∏dselsnummer()),
    tilrettelagtKommunikasjon: tilrettelagtKommunikasjonAremark,
    telefonnummer: [telefonAremark],
    kontaktOgReservasjon: digitalKontaktInformasjonAremark,
    bankkonto: getBankKonto()
};
