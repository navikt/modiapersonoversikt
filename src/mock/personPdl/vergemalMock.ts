import faker from 'faker/locale/nb_NO';

import navfaker from 'nav-faker';

import { getSistOppdatert, vektetSjanse } from '../utils/mock-utils';
import { Verge } from '../../models/personPdl/verge';
import { Navn } from '../../models/personPdl/person';

const VERGESAKSTYPER = [
    'ensligMindreaarigAsylsøker',
    'ensligMindreaarigFlyktning',
    'voksen',
    'midlertidigForVoksen',
    'mindreaarig',
    'midlertidigForMindreaarig',
    'forvaltningUtenforVergemaal',
    'stadfestetFremtidsfullmakt',
    null
];

const OMFANGTYPER = [
    'utlendingssakerPersonligeOgOekonomiskeInteresser',
    'personligeOgOekonomiskeInteresser',
    'oekonomiskeInteresser',
    'personligeInteresser'
];

export function mockVergemal(fodselsnummer: string): Verge[] {
    if (fodselsnummer === '10108000398') {
        return getAremarkVerge();
    }
    faker.seed(Number(fodselsnummer));
    navfaker.seed(fodselsnummer);
    if (vektetSjanse(faker, 0.7)) {
        return [];
    }
    return getVergemal();
}

function getVergemal() {
    const antallVerger = faker.random.number(3);
    let verger = [];
    for (let i = 0; i < antallVerger; i++) {
        verger.push(getVerge());
    }
    return verger;
}

function lagNavn(): Navn {
    return {
        fornavn: faker.name.firstName(),
        mellomnavn: null,
        etternavn: faker.name.lastName()
    };
}

function getVerge(): Verge {
    const vergesFodselsnummer = navfaker.personIdentifikator.myndigFødselsnummer();
    const vergemaalManglerVergeData = vektetSjanse(faker, 0.2);
    const navn: Navn | null = vergemaalManglerVergeData ? null : lagNavn();

    return {
        ident: vergemaalManglerVergeData ? null : vergesFodselsnummer,
        navn: navn,
        vergesakstype: getTilfeldigVergesakstype(),
        omfang: vektetSjanse(faker, 0.5) ? getTilfeldigOmfangtype() : null,
        embete: getTilfeldigEmbete(),
        gyldighetstidspunkt: getTilfeldigGyldighetstidspunkt(),
        opphoerstidspunkt: getTilfeldigOpphoersTidspunkt()
    };
}

function getTilfeldigVergesakstype(): string | null {
    const vergesakstype = VERGESAKSTYPER[faker.random.number(VERGESAKSTYPER.length - 1)];
    if (!vergesakstype) {
        return null;
    }
    return vergesakstype;
}

function getTilfeldigOmfangtype(): string | null {
    const omfang = OMFANGTYPER[faker.random.number(OMFANGTYPER.length - 1)];
    if (!omfang) {
        return null;
    }
    return omfang;
}

function getTilfeldigEmbete(): string {
    return 'Fylkesmannen i ' + faker.address.city();
}

function getTilfeldigGyldighetstidspunkt() {
    return vektetSjanse(faker, 0.8) ? new Date(getSistOppdatert()) : null;
}

function getTilfeldigOpphoersTidspunkt() {
    return vektetSjanse(faker, 0.8) ? new Date(getSistOppdatert()) : null;
}

function getAremarkVerge(): Verge[] {
    return [
        {
            ident: '21042900076',
            navn: {
                fornavn: 'Simen',
                mellomnavn: null,
                etternavn: 'Solli'
            },
            vergesakstype: 'voksen',
            omfang: 'personligeOgOekonomiskeInteresser',
            embete: 'Fylkesmannen i Troms og Finnmark',
            gyldighetstidspunkt: new Date('2016-03-27'),
            opphoerstidspunkt: new Date('2013-06-30')
        }
    ];
}
