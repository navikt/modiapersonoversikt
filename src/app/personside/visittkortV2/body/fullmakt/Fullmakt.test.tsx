import * as React from 'react';
import * as renderer from 'react-test-renderer';
import TestProvider from '../../../../../test/Testprovider';
import { aremark } from '../../../../../mock/personPdl/aremark';
import Fullmakter from './Fullmakt';

test('viser fullmakt', () => {
    const visittkortbody = renderer.create(
        <TestProvider>
            <Fullmakter fullmakter={aremark.fullmakt} />
        </TestProvider>
    );

    expect(visittkortbody.toJSON()).toMatchSnapshot();
});
