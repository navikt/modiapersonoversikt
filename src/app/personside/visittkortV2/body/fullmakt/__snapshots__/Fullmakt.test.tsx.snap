// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`viser fullmakt 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  position: absolute;
  left: -3.125rem;
  width: 3.125rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 > svg {
  height: 1.5rem;
  width: auto;
}

.c3 {
  margin: 10px 0 20px 0;
}

.c3:last-child {
  margin: 10px 0 0 0;
}

.c2 {
  margin-bottom: 1rem;
}

<section>
  <div
    className="c0"
  >
    <div
      className="c1"
    >
      <svg
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="50"
          cy="50"
          fill="#FFB249"
          r="50"
        />
        <line
          stroke="#3E3832"
          strokeLinecap="round"
          strokeWidth="4"
          x1="50"
          x2="50"
          y1="25"
          y2="57"
        />
        <circle
          cx="50"
          cy="72"
          fill="#3E3832"
          r="4.5"
        />
      </svg>
    </div>
    <h3
      className="typo-undertittel"
    >
      fullmakter
    </h3>
  </div>
  <div
    className="c2"
  />
  <div
    className="c3"
  >
    <div
      className="c0"
    >
      
      <h4
        className="typo-element"
      >
        Fullmektig
    : [object Object](123456789)
      </h4>
    </div>
    <p
      className="typo-normal"
    >
      Gyldig fra og med 
      01.01.2019
    </p>
    <p
      className="typo-normal"
    >
      Gyldig til og med 
      12.12.2022
    </p>
    <p
      className="typo-normal"
    >
      Gjelder 
      alle ytelser
    </p>
  </div>
</section>
`;
